### PR TITLE
BL-507 DS Feedback CRUD functions and Schema changes 

### PR DESCRIPTION
**Pull request by L0lbrich**
We created CRUD functions for the feedback class and modified the existing Feedback Schema in the Schema.py file to handle new data. 

READ: This function takes mentee_id, mentor_id, ticket_id, or None as arguments. Given mentee_id it will return all feedback for that mentee. Given mentor_id it will return all feedback for that mentor. Given ticket_id it will return the ticket with that ID. Given None, it will read all tickets.

CREATE: This function takes data in the format of the feedback schema. The arguments are Text, Ticket_id, mentee_id, mentor_id, DateTime, and vader_score. The Datetime, Ticket_id, and vader_score are all automatically generated upon submission of the ticket. The only fields you need to input are text, mentee_id, and mentor_id.

DELETE: This function takes ticket_id as an argument and deletes that ticket. It returns the deleted ticket

UPDATE: Update takes ticket_id as an argument and updates text, mentor_id, and mentee_id using the UpdateFeedback Schema. The Vader_score and DateTime functions will be autogenerated upon submission.

[Loom](https://www.loom.com/share/9918e550e48d4f98b0c2e2fdbba9308c)

Fixes # BL_507

## Type of change

- [x] New feature

## Checklist:

- [x] My code follows PEP8 style guide
- [x] I have removed unnecessary print statements from my code
- [x] I have made corresponding changes to the documentation if necessary
- [x] My changes generate no errors
- [x] No commented-out code
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by sspradling78 — DISMISSED** [review]
This will make #168 work

**Review by jsgersing — APPROVED** [review]
CRUD functions for Feedback look good and consistent with the associated Schema changes.

**Review by marthabg — APPROVED** [review]
Good call editing the feedback schema to add the text, datetime, and vader_score to work in tandem with your API. I'm not all too familiar with API endpoint code yet so I can't say much on that. I was wondering though if "Datetime" in your schema should be called something else since it is highlighted orange, leads me to believe it's not just a variable name, unless that's doing something special on purpose.

**Review by sujatasahoo — APPROVED** [review]
Good update of Feedback collection API and learned something new, CRUD function.

**Review by ErinNC — APPROVED** [review]
Hooray for separate CRUD endpoints! Only minor recommendation I see would be to match the pattern of mentor and mentee in schema.py by having Feedback collection above FeedbackUpdate.

**Review by ErinNC — DISMISSED** [review]
Schema file looks good!

**Review by rogerfvieira — DISMISSED** [review]
Very clear explanation and clean implementation of CRUD functionality to a collection.

**Review by sujatasahoo — DISMISSED** [review]
Good update on Feedback Schema and application of CRUD functions.

**Review by jsgersing — DISMISSED** [review]
I thought I had reviewed previously, but it wasn't registering.

**Review by jinjahninjah — CHANGES_REQUESTED** [review]
Great Job! Just a couple of things to fix, but otherwise solid work.

Also, check with Robert on the `__init__` method. He might have a better solution for ya 👍

**Comment by BrokenShell** [comment]
Please ignore codeclimate, it's being over sensitive. I think it just needs a hug, but I'm not an expert.

**Comment by woodbenwood** [comment]
New more soothing loom (for push updates 7/21):
https://www.loom.com/share/a273e45ab592467fb5063a3c935e3681

File: app/api.py [hunk 1]
```diff
@@ -209,6 +211,87 @@ async def sentiment(text: str):
...
+    mentees_df = pd.DataFrame(API.db.read("Mentees"))[["tech_stack", "name"]]
+    mentors_df["user_role"] = "Mentor"
+    mentees_df["user_role"] = "Mentee"
+    df = pd.concat([mentees_df, mentors_df], axis=0).reset_index(drop=True)
+    return json.loads(tech_stack_by_role(df).to_json())
+
+
+@API.get("/responses_analysis")
+async def responses_analysis():
+    """Returns relevant topics for analysis usage
```
Comment (BrokenShell) [inline app/api.py:227]: Let's get rid of this endpoint.

File: app/api.py [hunk 2]
```diff
@@ -209,6 +211,87 @@ async def sentiment(text: str):
     return {"result": vader_score(text)}
 
 
+@API.get("/graphs/tech-stack-by-role")
+async def tech_stack_graph():
```
Comment (BrokenShell) [inline app/api.py:215]: Let's get rid of this endpoint.

File: app/api.py [hunk 3]
```diff
@@ -209,6 +211,87 @@ async def sentiment(text: str):
...
+    """
+    return {"result": API.db.read("Feedback", query.dict())}
+
+
+@API.post("/create/feedback")
+async def create_feedback(data: Feedback):
+    """Create records in the feedback collection.
+
+        Creates new document within Feedback using the Feedback schema to
+        populate its fields. Creates ticket_id and vader_score automatically on submission
```
Comment (BrokenShell) [inline app/api.py:251]: This docstring suggests that a ticket_id will be created automatically. I'm pretty sure this is incorrect.

**Review by BrokenShell — CHANGES_REQUESTED** [review]

File: app/api.py (hunk 3 above)
Comment (woodbenwood) [reply app/api.py:251]: Correct. While not the most elegant thing I've written, I believe this edit dispels the misinformation: """"Create records in the feedback collection.
        Creates new document within Feedback using the Feedback schema to
        populate its fields, except for ticket_id, which comes from FE."

**Pull request merged**